internal/server/data/provideruser.go

```diff
@@ -49,3 +53,39 @@ func DeleteProviderUsers(db *gorm.DB, selectors ...SelectorFunc) error {
 func GetProviderUser(db *gorm.DB, providerID, userID uid.ID) (*models.ProviderUser, error) {
 	return get[models.ProviderUser](db, ByProviderID(providerID), ByIdentityID(userID))
 }
+
+func SyncProviderUser(ctx context.Context, db *gorm.DB, user *models.Identity, provider *models.Provider, oidcClient providers.OIDCClient) error {
```

Comment:
Moved the `SyncProviderUser` function to the data package to the data package to avoid a circular dependency during testing and migrations. In general it makes more sense for this part of the user sync to live here because it interacts with the database, and that should only happen in the data package.